dvc/remote/gdrive/__init__.py

```diff
@@ -0,0 +1,323 @@
+        item_list = self.drive.ListFile(
+            {"q": query, "maxResults": 1}
+        ).GetList()
+        return next(iter(item_list), None)
+
+    def resolve_remote_item_from_path(self, path_parts, create):
+        parents_ids = ["root"]
+        current_path = ""
+        for path_part in path_parts:
+            current_path = posixpath.join(current_path, path_part)
+            remote_ids = self.get_remote_id_from_cache(current_path)
+            if remote_ids:
+                parents_ids = remote_ids
+                continue
+            item = self.get_remote_item(path_part, parents_ids)
+            if not item and create:
+                item = self.create_remote_dir(parents_ids[0], path_part)
+            elif not item:
+                return None
+            parents_ids = [item["id"]]
+        return item
+
+    def get_remote_id_from_cache(self, remote_path):
+        if hasattr(self, "_cached_dirs"):
+            return self.cached_dirs.get(remote_path, [])
+        return []
+
+    def get_remote_id(self, path_info, create=False):
+        remote_ids = self.get_remote_id_from_cache(path_info.path)
+
+        if remote_ids:
+            return remote_ids[0]
+
+        file1 = self.resolve_remote_item_from_path(
+            path_info.path.split("/"), create
+        )
+        return file1["id"] if file1 else ""
+
+    def exists(self, path_info):
+        return self.get_remote_id(path_info) != ""
+
+    def _upload(self, from_file, to_info, name, no_progress_bar):
+        dirname = to_info.parent
+        if dirname:
+            parent_id = self.get_remote_id(dirname, True)
+        else:
+            parent_id = to_info.bucket
+
+        gdrive_retry(
+            lambda: self.gdrive_upload_file(
+                {"title": to_info.name, "parent_id": parent_id},
+                no_progress_bar,
+                from_file,
+                name,
+            )
+        )()
+
+    def _download(self, from_info, to_file, name, no_progress_bar):
+        file_id = self.get_remote_id(from_info)
+        gdrive_retry(
+            lambda: self.gdrive_download_file(
+                file_id, to_file, name, no_progress_bar
+            )
+        )()
+
+    def list_cache_paths(self):
+        file_id = self.get_remote_id(self.path_info)
+        prefix = self.path_info.path
+        for path in self.list_children(file_id):
+            yield posixpath.join(prefix, path)
+
+    def list_children(self, parent_id):
+        for file1 in self.gdrive_list_item(
+            "'{}' in parents and trashed=false".format(parent_id)
+        ):
+            for path in self.list_remote_item(file1):
+                yield path
+
+    def list_remote_item(self, drive_file):
+        if drive_file["mimeType"] == FOLDER_MIME_TYPE:
+            for i in self.list_children(drive_file["id"]):
+                yield posixpath.join(drive_file["title"], i)
+        else:
+            yield drive_file["title"]
+
+    def all(self):
```

Comment:
There is no reason to implement both this and `list_cache_paths()`.

Reply:
`list_cache_paths` should be removed for now. Resolving this.